internal/middleware/util.go

```diff
@@ -85,66 +85,42 @@ func ExtractStringField(data map[string]interface{}, key string) (string, error)
 }
 
 // ValidateCommonResourceData Validates the "commonResourceData" field using a predefined schema.
-func ValidateCommonResourceData(resourceType string, resource map[string]interface{}) error {
-
+func ValidateCommonData(resourceType string, commonData map[string]interface{}) error {
 	commonSchemaKey := fmt.Sprintf("common:%s", strings.ToLower(resourceType))
 	commonSchema, err := getSchemaFromCache(commonSchemaKey)
 	if err != nil {
 		return fmt.Errorf("failed to load common resource schema for '%s': %w", resourceType, err)
 	}
 
-	commonResourceData, exists := resource["commonResourceData"].(map[string]interface{})
-	if !exists {
-		return fmt.Errorf("missing required 'commonResourceData' for resource '%s'", resourceType)
-	}
-
-	if err := ValidateJSONSchema(commonSchema, commonResourceData); err != nil {
-		return fmt.Errorf("commonResourceData validation failed for '%s': %w", resourceType, err)
+	if err := ValidateJSONSchema(commonSchema, commonData); err != nil {
+		return fmt.Errorf("common resource data validation failed for '%s': %w", resourceType, err)
 	}
 
 	return nil
 }
 
-// Validates the reporter resource schema by checking if it exists and ensuring resourceData follows its structure.
-func ValidateReporterResourceData(resourceType string, reporterData map[string]interface{}) error {
-	// Extract reporterType from the provided data
-	reporterType, hasReporterType := reporterData["reporterType"].(string)
-	if !hasReporterType || reporterType == "" {
-		return fmt.Errorf("missing or invalid 'reporter_type' in reporterData for resource '%s'", resourceType)
-	}
-
-	// Construct the schema key using the new format: resourceType:reporterType
+// Validates the reporter-specific data against its schema based on resourceType and reporterType.
+func ValidateReporterData(resourceType string, reporterType string, reporterData map[string]interface{}) error {
```

Comment:
another example of `Representation` vs `Data`

Reply:
Updated